packages/gatsby/src/redux/actions/add-page-dependency.js

```diff
@@ -24,7 +23,7 @@ function createPageDependency({ path, nodeId, connection }) {
   }
   if (
     connection &&
-    _.includes(state.componentDataDependencies.connections, connection)
+    state.componentDataDependencies.connections.has(connection)
```

Comment:
Ok, this is actually causing some problems - in this check we never do any `path` check (and this is broken in master as well, but in master the condition actually never is true, so we might do some extra work, but it doesn't actually break data tracking.

With this change this become broken - we should use same approach as in `nodeId` check above:
```suggestion
    state.componentDataDependencies.connections.has(connection) &&
    state.componentDataDependencies.connections.get(connection).has(path)
```

Reply:
Done